src/resource/wacAllow.internal.ts

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright 2021 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+// TODO: Should we move this type internally, as the Access in this file is not strictly the same as ACL Access
+import { Access } from "../acl/acl";
+
+export type internal_AccessPermissions = {
+  user: Access;
+  public: Access;
+};
+
+interface internal_parsedWacHeader {
+  [group: string]: string[];
+}
+
+function internal_parseWacHeader(header: string): internal_parsedWacHeader {
+  const rawEntries = header.split(",").map((rawEntry) => rawEntry.split("="));
+
+  const entries: internal_parsedWacHeader = {};
+
+  for (const rawEntry of rawEntries) {
+    if (rawEntry.length !== 2) {
+      continue;
+    }
+
+    const scope = rawEntry[0].trim();
+    const statement = rawEntry[1].trim();
+
+    if (!scope || !statement) {
+      continue;
+    }
+
+    if (
+      statement.length === 2 ||
+      statement.charAt(0) !== '"' ||
+      statement.charAt(statement.length - 1) !== '"'
+    ) {
+      continue;
+    }
+
+    const accessModes = statement.substring(1, statement.length - 1).trim();
+
+    if (accessModes.length === 0) {
+      continue;
+    }
+
+    entries[scope] = accessModes.split(" ");
+  }
+
+  return entries;
+}
+
+function internal_getWacPermissions(
```

Comment:
Not sure this is named correctly / strictly belongs here; perhaps `internal_getPermissions` would be better, as it's just checking a parsed WAC Allow for given values

Reply:
Sounds good to me, we can always move it if it becomes relevant elsewhere.